### ISSUE TITLE
Error if folder doesn't exist when using `fallbackLng`

### ISSUE THREAD

**Issue by aaimio** [i18next/next-i18next#1941]
## 🐛 Bug Report

- We run multiple domains, some of which are using the same `defaultLocale`
- Next doesn't support specifying the same `defaultLocale` for multiple domains
  - https://github.com/vercel/next.js/discussions/37991
  - https://stackoverflow.com/questions/73410976/next-js-i18n-2-domains-with-the-save-default-locale 
- To get past this, we specify unique variants of languages e.g. English (`en-DE`, `en-EN`) and rely on `fallbackLng`  to serve the correct language using the configuration below:

**`next-i18next.config.js`**

```JS
module.exports = {
  i18n: {
    defaultLocale: "en",
    locales: ["en", "en-DE", "en-EN"],
  },
  fallbackLng: {
    default: ["en"],
    "en-DE": ["en"],
  },
  debug: process.env.NODE_ENV === 'development',
  reloadOnPrerender: process.env.NODE_ENV === 'development',
};

```

**`next.config.js`**

```JS
const { i18n } = require("./next-i18next.config");

/** @type {import('next').NextConfig} */
const nextConfig = {
  reactStrictMode: true,
  swcMinify: true,
  i18n: {
    ...i18n,
    localeDetection: false,
    domains: [
      {
        domain: "next-i18next.com",
        defaultLocale: "en-EN",
        locales: ["en-EN"],
      },
      {
        domain: "de.next-i18next.com",
        defaultLocale: "en-DE",
        locales: ["en-DE"],
      },
    ],
  },
};

module.exports = nextConfig;

```

The above works, but only if the folder for that fallback language exists, if the folder doesn't exist we run into this error:

![oops](https://user-images.githubusercontent.com/44427668/187875901-ed10a87e-d76d-48ad-b70e-3796d386914f.png)

It seems that we're not checking whether the locale folder exists before we attempt to read from it:

https://github.com/i18next/next-i18next/blob/master/src/serverSideTranslations.ts#L80-L88

## To Reproduce

- Checkout this repo featuring the issue: https://github.com/aaimio/next-i18next-bug
- Set up two domains in your `/etc/hosts`
  - `127.0.0.1 de.next-i18next.com`
  - `127.0.0.1 next-i18next.com`
- `npm install`
- `npm run dev`
- In your browser, visit `de.next-i18next.com` you will get the same error

## Expected behavior

We should first check whether the folder exists before reading from it.

- If default locale is specified, and its folder **doesn't exist**, fallback to value in `fallbackLng`, use that language's JSON file
- If default locale is specified, and its folder **exists**, use that language's JSON file

## Your Environment

- *Runtime version*: v18.4.0
- *next-i18next version*: 12.0.0
- *OS*: Mac

**Comment by adrai** [comment]
@chrilis do you like to have a look at this?

**Comment by aaimio** [comment]
This seems to do the trick, but I'm not sure how the tests should be updated.

```diff
const getLocaleNamespaces = (path: string) =>
- fs.readdirSync(path)
+ fs.existsSync(path) ? fs.readdirSync(path)
    .map(file => file.replace(`.${localeExtension}`, ''))
```

**Comment by chrilis** [comment]
@aaimio The reason for the error is that no namespace is specified in `getServerSideTranslations`. If you need a quick fix, you could define it manually, eg. ` ...(await serverSideTranslations(locale, ['common']))`. I'll try to fix the bug and adjust the tests.

**Comment by aaimio** [comment]
> @aaimio The reason for the error is that no namespace is specified in `getServerSideTranslations`. If you need a quick fix, you could define it manually, eg. ` ...(await serverSideTranslations(locale, ['common']))`. I'll try to fix the bug and adjust the tests.

Ah that makes sense, currently I create empty folders to workaround this, but I can see that passing the namespace explicitly will bypass this check where the code errors.

Appreciate the quick reply and look forward to the fix!

**Comment by adrai** [comment]
should be better with v12.0.1

**Comment by chrilis** [comment]
I think there is a typo in the release description: #1941 is closed, not #1302.

**Comment by adrai** [comment]
uups...
should be fixed now
